File: day01/task01.py
import io, re

file = open("input.txt", "r")

sum = 0

for line in file.readlines():
    try:
        x = re.search("(\d).*(\d)", line).groups()
    except:
        x = re.search("(\d)", line).groups()
    if len(x) == 1:
        number = int(x[0]) * 10 + int(x[0])
    else:
        number = int(x[0]) * 10 + int(x[1])
    sum += number
print(sum)
file.close()

File: day15/task01.rs
use std::fs::File;
use std::io::{self, Read};

fn main() -> io::Result<()> {
    let file_path = "input.txt";
    let mut file = File::open(file_path)?;
    let mut buffer = String::new();

    file.read_to_string(&mut buffer)?;

    let parts = buffer.split(",");

    let mut sum = 0;

    for part in parts {
        let mut temp_sum = 0 as u32;

        for char in part.chars() {
            let ascii = char as u8;
            temp_sum += ascii as u32;
            temp_sum *= 17;
            temp_sum %= 256;
        }
        sum += temp_sum;
    }

    println!("END... {}", sum);

    Ok(())
}
